ctapipe/io/tableloader.py

```diff
@@ -0,0 +1,330 @@
+                keys=["tel_id"],
+                join_type="inner",
+            )
+        
+        # Instrument information doesn't depend on the triggered tel_ids
+        # but depends on the tel_ids for joining
+        if self.load_instrument and ((table is None) or (len(table)==0)):
+            table = self.instrument_table
+        elif self.load_instrument and len(table) > 0:
+            table = join(
+                table, self.instrument_table, keys=["tel_id"], join_type="inner"
+            )
+        
+        # DL2 data (either simulated or reconstructed) depends only on
+        # event_id(s), not on any specific tel_id(s)
+        # so we add it at the end
+        if self.load_dl2_geometry:
+            
+            # get reconstructors in file (if more than one)
+            reconstructors = self._get_reconstructors()
+                        
+            for i, reconstructor in enumerate(reconstructors):
+                
+                geometry = self._read_telescope_DL2_geometry(GEOMETRY_GROUP, reconstructor)
+            
+                # rename DL2 columns to explicit reconstructor
+                # TBD: we could skip this if only 1 reconstructor is present
+                # or simply find another way to deal with multiple reconstructions
+                for col in set(geometry.colnames) - {"obs_id", "event_id"}:
+                    geometry.rename_column(col, f"{reconstructor}_{col}")
+                
+                if (table is None or len(table)==0) and i==0:
+                    table = geometry
+                else:
+                    table = join(
+                        table, geometry, keys=["obs_id", "event_id"], join_type="left"
+                    )
+
+        if self.shower_table and (table is None or len(table)==0):
+            table = self.shower_table
+        elif self.shower_table and (table is not None):
+            table = join(
+                table, self.shower_table, keys=["obs_id", "event_id"], join_type="left"
+            )
+        
+        # Trigger information doesn't depend on the triggered tel_ids
+        # but depends on the "obs_id", "event_id" for joining
+        if self.load_trigger and ((table is None) or (len(table)==0)):
+            table = self.trigger_table
+        elif self.load_trigger and len(table) > 0:
+            table = join(
+                table,
+                self.trigger_table,
+                keys=["obs_id", "event_id"],
+                join_type="inner",
+            )
+
+        return table
+
+    def read_telescope_events_for_type(self, tel_type):
+        tel_ids = self.subarray.get_tel_ids_for_type(tel_type)
+        return self.read_telescope_events(tel_ids)
+
+    def read_telescope_events_for_id(self, tel_id, table=None):
+
+        if self.load_dl1_parameters:
+            table = self._read_telescope_table(PARAMETERS_GROUP, tel_id)
```

Comment:
This overwrites `table` without checking if it is None. For what is the `table` argument?

Reply:
removed